core/src/main/java/org/elasticsearch/search/slice/Slice.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.slice;
+
+import java.util.Objects;
+
+/**
+ *
+ */
+public final class Slice {
+    private final String field;
+    // The id of the slice.
+    private final int id;
+    // The total number of slices.
+    private final int max;
+
+    public Slice(String field, int id, int max) {
+        this.field = field;
+        this.id = id;
+        this.max = max;
+    }
+
+    public String getField() {
+        return field;
+    }
+
+    public int getId() {
+        return this.id;
+    }
+
+    public int getMax() {
+        return this.max;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(this.field, this.id, this.max);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other instanceof Slice) {
+            Slice o = (Slice) other;
+            return (field.equals(o.field) && id == o.id && o.max == max);
+        }
+        return false;
+    }
+
+    /**
+     * Returns true if the value is contained within the slice
+     */
+    public boolean contains(long value) {
+        if ((abs(value) % max) == id) {
```

Comment:
maybe we should do Math.floorMod(value, max)? This way negative values and Long.MIN_VALUE in particular are not an issue